### PR TITLE
DW-4689 - Migrate reverse-proxy to ECS container from EC2

### PR DESCRIPTION
**Pull request by benjaminsherwood**
Signed-off-by: Benjamin Sherwood <benjaminsherwood@digital.uc.dwp.gov.uk>

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -18,3 +18,7 @@ variable "ssh_bastion_ami_id" {
   type    = string
   default = ""
 }
+
+variable "reverse_proxy_http_port" {
+  default = 80
+}
```
Comment (ChrisScottThomas) [inline variables.tf]:
```suggestion
}

```

**Review by ChrisScottThomas — DISMISSED** [review]

File: container-reverse-proxy.tf [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+    ]
+  }
+]
+DEFINITION
+
+}
+
+resource "aws_ecs_service" "container_internet_proxy" {
+  count           = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  name            = "container-internet-proxy"
```
Comment (tea-short) [inline container-reverse-proxy.tf]: Would it be better/clearer if this was "reverse-proxy"? Maybe worth defining a local, e.g. 
`reverse_proxy_name = "reverse-proxy"` and then referring to it everywhere the name is needed, like here and in cloudwatch log group name?

Comment (benjaminsherwood) [reply container-reverse-proxy.tf]: yea it's meant to be :P

File: container-reverse-proxy.tf [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+}
+
+resource "aws_security_group_rule" "reverse_proxy_http_ingress" {
+  count             = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  description       = "Reverse Proxy Container HTTP Rule"
+  type              = "ingress"
+  protocol          = "tcp"
+  from_port         = "80"
+  to_port           = "80"
+  cidr_blocks       = [module.vpc.vpc.cidr_block]
```
Comment (tea-short) [inline container-reverse-proxy.tf:283]: Could we tighten this down? Isn't it just VPN or DirectConnect where the connections would originate from?

Comment (benjaminsherwood) [reply container-reverse-proxy.tf:283]: No, it's a LB to ECS rule?

Comment (benjaminsherwood) [reply container-reverse-proxy.tf:283]: Actually, if this doesn't have a LB SG now you may be right. ECS is new to me.

File: container-reverse-proxy.tf [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+  cpu                      = "512"
+  memory                   = "4096"
+  task_role_arn            = aws_iam_role.container_reverse_proxy[0].arn
+  execution_role_arn       = data.terraform_remote_state.management.outputs.ecs_task_execution_role.arn
+
+  container_definitions = <<DEFINITION
+[
+  {
+    "image": "${local.account[local.environment]}.${module.vpc.ecr_dkr_domain_name}/nginx-s3:latest",
+    "name": "squid-s3",
```
Comment (mattburgess) [inline container-reverse-proxy.tf]:
```suggestion
    "name": "nginx-s3",
```

**Review by mattburgess — APPROVED** [review]

**Pull request merged**